server/routes/api/index.js

```diff
@@ -1,4 +1,123 @@
 const express = require('express');
+const bcrypt = require('bcryptjs');
+const passport = require('passport');
+const saltRounds = 12;
+
+const User = require('../../../database/models/User');
+
 const router = express.Router();
 
+function isAuthenticated(req, res, next) {
+  if (req.isAuthenticated()) {
+    next();
+  } else {
+    res.redirect('/');
+  }
+}
+
+router.get('/profile', isAuthenticated, (req, res) => {
+  const userId = req.user.id;
+
+  User.where('id', userId)
+    .fetch()
+    .then(dbUser => {
+      return res.json(dbUser);
+    })
+    .catch(err => {
+      res.status(500);
+      return res.json({
+        success: false,
+        error: err
+      });
+    });
+});
+
+router.put('/users', isAuthenticated, (req, res) => {
+  const userId = req.user.id;
+  const { username, name, email, address } = req.body;
+  const editedUserReq = { username, name, email, address };
+
+  User.where('id', userId)
+    .fetch()
+    .then(dbUser => {
+      dbUser
+        .save(editedUserReq)
+        .then(resUser => {
```

Comment:
You should return the dbUser, then the `.then` and the `.catch` here can be part of the upper promise chain.